source/agora/consensus/protocol/Nominator.d

```diff
@@ -541,14 +541,8 @@ extern(D):
                     block_sig.signature, block_sig.public_key, block_sig.height);
         if (cur_height >= block_sig.height)
         {
-            auto blocks = this.ledger.getBlocksFrom(Height(block_sig.height));
-            if (blocks.empty)
-            {
-                log.warn("Current block height is {}. Block for slot {} not found",
-                    cur_height, block_sig.height);
-                return;
-            }
-            const Block block = blocks.front;
+            const block = this.ledger.getBlocksFrom(Height(block_sig.height))
+                .front;
```

Comment:
The irony here is that the storage will throw an exception if it can't read that height. So the `blocks.empty` check again didn't do anything because it was too late.